components/Snackbar/src/MDCSnackbarManager.m

```diff
@@ -498,19 +491,49 @@ - (void)removeSuspensionIdentifierMainThread:(NSUUID *)identifier forCategory:(N
 
 #pragma mark - Public API
 
-@implementation MDCSnackbarManager
+@interface MDCSnackbarManager ()
+@property(nonnull, nonatomic, strong) MDCSnackbarManagerInternal *internalManager;
+@end
 
-+ (void)setDelegate:(id<MDCSnackbarManagerDelegate>)delegate {
-  MDCSnackbarManagerInternal *manager = [MDCSnackbarManagerInternal sharedInstance];
-  manager.delegate = delegate;
+@implementation MDCSnackbarManager {
+  UIColor *_snackbarMessageViewBackgroundColor;
+  UIColor *_snackbarMessageViewShadowColor;
+  UIColor *_messageTextColor;
+  UIFont *_messageFont;
+  UIFont *_buttonFont;
+  NSMutableDictionary<NSNumber *, UIColor *> *_buttonTitleColors;
+  BOOL _mdc_adjustsFontForContentSizeCategory;
+  BOOL _shouldApplyStyleChangesToVisibleSnackbars;
 }
 
-+ (id<MDCSnackbarManagerDelegate>)delegate {
-  MDCSnackbarManagerInternal *manager = [MDCSnackbarManagerInternal sharedInstance];
-  return manager.delegate;
+
++ (instancetype)defaultManager {
+  static MDCSnackbarManager *kDefaultManager;
+  static dispatch_once_t onceToken;
+  dispatch_once(&onceToken, ^{
+    kDefaultManager = [[MDCSnackbarManager alloc] init];
+  });
+  return kDefaultManager;
 }
 
-+ (void)showMessage:(MDCSnackbarMessage *)inputMessage {
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _internalManager = [[MDCSnackbarManagerInternal alloc] initWithSnackbarManager:self];
```

Comment:
does this need to pass `self`? the logic is a bit hard to follow in the PR, so just verifying this is intended.

Reply:
Was also curious about this. Had wondered if a delegate pattern might be more clear here as well, but couldn't convince myself that it would substantially improve the quality of the implementation.

@romoore I'm assuming we'll be able to remove this cyclical reference after we remove the old static APIs?